modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactoryBean.java

```diff
@@ -62,6 +62,6 @@ protected IgniteRepositoryFactoryBean(Class<? extends T> repoInterface) {
 
     /** {@inheritDoc} */
     @Override protected RepositoryFactorySupport createRepositoryFactory() {
-        return new IgniteRepositoryFactory(ctx);
+        return new IgniteRepositoryFactory(ctx, getObjectType());
```

Comment:
There is a one-to-one relation between `IgniteRepositoryFactoryBean` and `IgniteRepositoryFactory`, also there is a one-to-one relation between `IgniteRepositoryFactory` and `IgniteProxy`. Perhaps we can simplify a little (but I'm not sure) if we mark `IgniteRepositoryFactoryBean` as disposable, and close factory and Ignite proxy on bean destroy. In this case, we don't need `IgniteProxyFactory` class, `igniteProxyFactory` and `igniteProxy` beans.
WDYT?

Reply:
I see the following problem: IgniteRepositoryFactory is created during IgniteRepositoryFactoryBean bean initialization. Therefore, if the initialization of the IgniteRepositoryFactoryBean fails with an error (for example, in case any query method has invalid configuration or some Spring bean internal dependencies issues), Spring will not call the "destroy" method, and since the Ignite instance was started earlier, it remains unclosed.